tests/test_transformations.py

```diff
@@ -0,0 +1,177 @@
+                         [((10, 10), (6, 6)),
+                          ((10, 10), (12, 12)),
+                          ((510, 510), (10, 10)),
+                          ((10, 510), (10, 10)),
+                          ((510, 10), (10, 10))])
+def test_video_subset_coord_pair(coordinate_pair, window_size):
+    current_path = (Path(__file__)).parent
+    video_path = current_path / 'resources' / 'test_video.h5'
+    with h5py.File(video_path, 'r') as open_h5:
+        video = open_h5['data'][:]
+        transformed_coordinates = transformations.get_transformed_center(coordinate_pair,
+                                                                         window_size,
+                                                                         video.shape)
+        left_column = transformed_coordinates[0] - math.ceil(window_size[0] / 2)
+        right_column = transformed_coordinates[0] + math.ceil(window_size[0] / 2)
+        up_row = transformed_coordinates[1] - math.ceil(window_size[1] / 2)
+        down_row = transformed_coordinates[1] + math.ceil(window_size[1] / 2)
+
+        # transform video
+        transformed_video = transformations.get_centered_coordinate_box(coordinate_pair,
+                                                                        window_size,
+                                                                        video)
+        # random sample frame to test subset
+        random_index = random.randint(0, 50)
+        random_transformed_frame = transformed_video[random_index]
+        random_regular_frame = video[random_index]
+        regular_subset = random_regular_frame[up_row:down_row,
+                                              left_column:right_column]
+
+        subtraction_frame = random_transformed_frame - regular_subset
+        assert not subtraction_frame.any()
+
+
+def test_mp4_conversion(video_fixture):
+    with h5py.File(video_fixture, 'r') as open_video:
+        video = open_video['data'][:]
+
+        norm_video = transformations.normalize_video(video)
+        output_path = video_fixture.parent
+        output_path = output_path / 'video.mp4'
+
+        transformations.transform_to_mp4(video=video,
+                                         output_path=output_path.as_posix())
+
+        frames = []
+        reader = imageio.get_reader(output_path.as_posix(), mode='I', fps=30,
+                                    size=(500, 500), pixelformat="gray")
+        for frame in reader:
+            gray_frame = cv2.cvtColor(frame, cv2.COLOR_RGB2GRAY)
+            frames.append(gray_frame)
+        frames = np.stack(frames)
+
+        # codec for mp4 encoding changes the video slightly between h5 and mp4
+        # and vice versa, so numbers are not exactly the same. Instead general
+        # structure of video array loaded. Cv color conversion does not rectify
+        # this error unfortunately, this has been attempted and small errors
+        # persist
```

Comment:
How small? Small enough for `numpy.testing.assert_close` etc?

Reply:
maybe...

Reply:
I don't think so, it comes from transitioning from grey to colored to back since mp4 requires rgb writing and reads out as rgb as well.

Reply:
looks like this comment is outdated, but, `numpy.allclose` and `numpy.testing.assert_allclose` have parameters for adjusting required relative and absolute tolerances for cases where default machine precision is too stringent.
https://numpy.org/doc/stable/reference/generated/numpy.testing.assert_allclose.html